pulser-core/pulser/noise_model.py

```diff
@@ -37,51 +39,94 @@
     "eff_noise",
 ]
 
-
-@dataclass(frozen=True)
+_NOISE_TYPE_PARAMS = {
+    "doppler": ("temperature",),
+    "amplitude": ("laser_waist", "amp_sigma"),
+    "SPAM": ("p_false_pos", "p_false_neg", "state_prep_error"),
+    "dephasing": ("dephasing_rate", "hyperfine_dephasing_rate"),
+    "relaxation": ("relaxation_rate",),
+    "depolarizing": ("depolarizing_rate",),
+    "eff_noise": ("eff_noise_rates", "eff_noise_opers"),
+}
+
+_PARAM_TO_NOISE_TYPE = {
+    param: noise_type
+    for noise_type, params in _NOISE_TYPE_PARAMS.items()
+    for param in params
+}
+
+# Parameter characterization
+
+_POSITIVE = {
+    "dephasing_rate",
+    "hyperfine_dephasing_rate",
+    "relaxation_rate",
+    "depolarizing_rate",
+}
+_STRICT_POSITIVE = {
+    "runs",
+    "samples_per_run",
+    "temperature",
+    "laser_waist",
+}
+_PROBABILITY_LIKE = {
+    "state_prep_error",
+    "p_false_pos",
+    "p_false_neg",
+    "amp_sigma",
+}
+
+_LEGACY_DEFAULTS = {
+    "runs": 15,
+    "samples_per_run": 5,
+    "state_prep_error": 0.005,
+    "p_false_pos": 0.01,
+    "p_false_neg": 0.05,
+    "temperature": 50.0,
+    "laser_waist": 175.0,
+    "amp_sigma": 5e-2,
+    "relaxation_rate": 0.01,
+    "dephasing_rate": 0.05,
+    "hyperfine_dephasing_rate": 1e-3,
+    "depolarizing_rate": 0.05,
+}
+
+
+@dataclass(init=False, repr=False, frozen=True)
 class NoiseModel:
     """Specifies the noise model parameters for emulation.
 
-    Select the desired noise types in `noise_types` and, if necessary,
-    modifiy the default values of related parameters.
-    Non-specified parameters will have reasonable default values which
-    are only taken into account when the related noise type is selected.
+    Supported noise types:
+
+    - **relaxation**: Noise due to a decay from the Rydberg to
+      the ground state (parametrized by ``relaxation_rate``),
+      commonly characterized experimentally by the T1 time.
+    - **dephasing**: Random phase (Z) flip (parametrized
+      by ``dephasing_rate``), commonly characterized
+      experimentally by the T2* time.
+    - **depolarizing**: Quantum noise where the state is
+      turned into the maximally mixed state with rate
+      ``depolarizing_rate``. While it does not describe a
+      physical phenomenon, it is a commonly used tool to test
+      the system under a uniform combination of phase flip (Z) and
+      bit flip (X) errors.
+    - **eff_noise**: General effective noise channel defined by the
+      set of collapse operators ``eff_noise_opers`` and the
+      corresponding rates distribution ``eff_noise_rates``.
+    - **doppler**: Local atom detuning due to termal motion of the
+      atoms and Doppler effect with respect to laser frequency.
+      Parametrized by the ``temperature`` field.
```

Comment:
Won't that force me to explain what the other parameters in the formula are too?

Reply:
Yeah, I thought about that and I guess it would be too cumbersome... I thought of pointing to an article, but can't find an article with the `$k_{eff}$` https://opg.optica.org/oe/fulltext.cfm?uri=oe-30-14-24261&id=477110 or in the bible https://steck.us/alkalidata/rubidium87numbers.1.6.pdf